coordinator-common/src/main/java/org/apache/kafka/coordinator/common/runtime/CoordinatorRuntime.java

```diff
@@ -769,6 +772,7 @@ private void flushCurrentBatch() {
                     for (DeferredEvent event : currentBatch.deferredEvents) {
                         deferredEventQueue.add(offset, event);
                     }
+                    runtimeMetrics.recordFlushTime(time.milliseconds() - flushStartMs);
```

Comment:
nit: Should we record this just before `updateLastWrittenOffset`? This is actually what you did on the transaction completion path.